src/test/java/com/waverleysoftware/data/DataSuppliers.java

```diff
@@ -32,6 +33,20 @@ public StreamEx getSensorData() {
                 new Sensor(3, "Garage", EMBEDDED)));
     }
 
+    @SuppressWarnings("unchecked")
+    @DataSupplier(transpose = true)
+    public <T> T[] getDataCollection(final Method method) {
+        final Optional<Data[]> annotationsByType = ofNullable(method.getAnnotationsByType(Data.class));
+        if (!annotationsByType.isPresent()) {
+            throw new NoClassDefFoundError("Data class not found");
+        }
+        return (T[]) StreamEx.of(annotationsByType)
```

Comment:
that's weird as you're wrapping optional with stream and perform some redundant flat mapping... you need to revise this part...